Index: Essentials/src/com/earth2me/essentials/commands/Commandtogglejail.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Essentials/src/com/earth2me/essentials/commands/Commandtogglejail.java	(revision 96590a338937ebfd1f580465c8db5619c1c2878e)
+++ Essentials/src/com/earth2me/essentials/commands/Commandtogglejail.java	(date 1589255917547)
@@ -4,6 +4,7 @@
 import com.earth2me.essentials.User;
 import com.earth2me.essentials.utils.DateUtil;
 import net.ess3.api.events.JailStatusChangeEvent;
+import org.bukkit.ChatColor;
 import org.bukkit.Server;
 
 import java.util.ArrayList;
@@ -43,23 +44,30 @@
             ess.getServer().getPluginManager().callEvent(event);
 
             if (!event.isCancelled()) {
-                if (player.getBase().isOnline()) {
-                    ess.getJails().sendToJail(player, args[1]);
-                } else {
-                    // Check if jail exists
-                    ess.getJails().getJail(args[1]);
-                }
-                player.setJailed(true);
-                player.sendMessage(tl("userJailed"));
-                player.setJail(null);
-                player.setJail(args[1]);
-                long timeDiff = 0;
-                if (args.length > 2) {
-                    final String time = getFinalArg(args, 2);
-                    timeDiff = DateUtil.parseDateDiff(time, true);
-                    player.setJailTimeout(timeDiff);
-                }
-                sender.sendMessage((timeDiff > 0 ? tl("playerJailedFor", player.getName(), DateUtil.formatDateDiff(timeDiff)) : tl("playerJailed", player.getName())));
+                double distanceToPlayer = controller.getBase().getLocation().distance(player.getBase().getLocation());
+                int maxRange = 25;
+                if (distanceToPlayer < maxRange || controller.isAuthorized("essentials.togglejail.rangeoverride")) {
+                    if (player.getBase().isOnline()) {
+                        ess.getJails().sendToJail(player, args[1]);
+                    } else {
+                        // Check if jail exists
+                        ess.getJails().getJail(args[1]);
+                    }
+                    player.setJailed(true);
+                    player.sendMessage(tl("userJailed"));
+                    player.setJail(null);
+                    player.setJail(args[1]);
+                    long timeDiff = 0;
+                    if (args.length > 2) {
+                        final String time = getFinalArg(args, 2);
+                        timeDiff = DateUtil.parseDateDiff(time, true);
+                        player.setJailTimeout(timeDiff);
+                    }
+                    sender.sendMessage((timeDiff > 0 ? tl("playerJailedFor", player.getName(), DateUtil.formatDateDiff(timeDiff)) : tl("playerJailed", player.getName())));
+                } else {
+                    controller.sendMessage(ChatColor.RED + "You must be closer to the player to jail them.");
+                }
+
             }
             return;
         }
Index: Essentials/src/com/earth2me/essentials/commands/Commandinvsee.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Essentials/src/com/earth2me/essentials/commands/Commandinvsee.java	(revision 96590a338937ebfd1f580465c8db5619c1c2878e)
+++ Essentials/src/com/earth2me/essentials/commands/Commandinvsee.java	(date 1589255786706)
@@ -1,6 +1,7 @@
 package com.earth2me.essentials.commands;
 
 import com.earth2me.essentials.User;
+import org.bukkit.ChatColor;
 import org.bukkit.Server;
 import org.bukkit.inventory.Inventory;
 
@@ -21,16 +22,22 @@
 
         final User invUser = getPlayer(server, user, args, 0);
         Inventory inv;
-
-        if (args.length > 1 && user.isAuthorized("essentials.invsee.equip")) {
-            inv = server.createInventory(invUser.getBase(), 9, "Equipped");
-            inv.setContents(invUser.getBase().getInventory().getArmorContents());
-        } else {
-            inv = invUser.getBase().getInventory();
-        }
-        user.getBase().closeInventory();
-        user.getBase().openInventory(inv);
-        user.setInvSee(true);
+        double distanceToTarget = user.getBase().getLocation().distance(invUser.getBase().getLocation());
+        int maxRange = 25;
+        if (distanceToTarget < 25 || user.isAuthorized("essentials.invsee.rangeoverride")) {
+            if (args.length > 1 && user.isAuthorized("essentials.invsee.equip")) {
+                inv = server.createInventory(invUser.getBase(), 9, "Equipped");
+                inv.setContents(invUser.getBase().getInventory().getArmorContents());
+            } else {
+                inv = invUser.getBase().getInventory();
+            }
+            user.getBase().closeInventory();
+            user.getBase().openInventory(inv);
+            user.setInvSee(true);
+        } else {
+            user.sendMessage(ChatColor.RED + "You need to be closer to check their pockets.");
+        }
+
     }
 
     @Override
